cumulus/parachains/runtimes/bridge-hubs/bridge-hub-rococo/src/xcm_config.rs

```diff
@@ -571,3 +585,114 @@ impl<
 		fee
 	}
 }
+
+/// A `HandleFee` implementation that takes fees from `ExportMessage` XCM instructions
+/// to Snowbridge and holds it in a receiver account. Burns the fees in case of a failure.
+pub struct XcmExportFeeToSnowbridge<
+	TokenLocation,
+	EthereumNetwork,
+	ReceiverAccount,
+	SovereignAccountOf,
+	AssetTransactor,
+	OutboundQueue,
+>(
+	PhantomData<(
+		TokenLocation,
+		EthereumNetwork,
+		ReceiverAccount,
+		SovereignAccountOf,
+		AssetTransactor,
+		OutboundQueue,
+	)>,
+);
+
+impl<
+		TokenLocation: Get<MultiLocation>,
+		EthereumNetwork: Get<NetworkId>,
+		ReceiverAccount: Get<AccountId>,
+		SovereignAccountOf: ConvertLocation<AccountId>,
+		AssetTransactor: TransactAsset,
+		OutboundQueue: OutboundQueueLocalFee<Balance = bp_rococo::Balance>,
+	> HandleFee
+	for XcmExportFeeToSnowbridge<
+		TokenLocation,
+		EthereumNetwork,
+		ReceiverAccount,
+		SovereignAccountOf,
+		AssetTransactor,
+		OutboundQueue,
+	>
+{
+	fn handle_fee(
+		fees: MultiAssets,
+		context: Option<&XcmContext>,
+		reason: FeeReason,
+	) -> MultiAssets {
+		let token_location = TokenLocation::get();
+		let mut fees = fees.into_inner();
+
+		if matches!(reason, FeeReason::Export { network: bridged_network, destination }
+				if bridged_network == EthereumNetwork::get() && destination == Here)
+		{
+			log::info!(
+				target: "xcm::fees",
+				"XcmExportFeeToSnowbridge fees: {fees:?}, context: {context:?}, reason: {reason:?}",
+			);
+
+			let fee_item_index = fees.iter().position(|asset| {
+				matches!(
+					asset,
+					MultiAsset { id: Concrete(location), fun: Fungible(..)}
+						if *location == token_location,
+				)
+			});
+			// Find the fee asset.
+			let fee_item = if let Some(element) = fee_item_index {
+				fees.remove(element)
+			} else {
+				return fees.into()
+			};
+
+			let receiver = ReceiverAccount::get();
+			// There is an origin so split fee into parts.
+			if let Some(XcmContext { origin: Some(origin), .. }) = context {
+				if let Some(origin) = SovereignAccountOf::convert_location(origin) {
```

Comment:
```suggestion
				if let Some(account) = SovereignAccountOf::convert_location(origin) {
```

Reply:
Addressed in https://github.com/Snowfork/polkadot-sdk/pull/23/commits/4e2b4ffd3a073289fe0d6ef3aa867450db65bbe2